### PR TITLE
#182201734 ft-most-travel-destinations: allow users to retrieve most travelled destinations

### PR DESCRIPTION
**Pull request by uwambajeddy**
#### What does this PR do?
This PR allow users to retrieve most travelled destinations

#### How should this be manually tested?
Trip request must have been approved to be noted in most travelled destinations
- `GET` `/api/v1/user/trip//most-travelled-destinations` for getting most travelled destinations
- for swagger doc use the following endpoint
Screenshot (370)

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/182201734

### CODE REVIEW

File: src/controllers/tripRequestController.js [hunk 1]
```diff
@@ -238,3 +239,44 @@ export const deleteTripRequest = async (req, res) => {
     return res.status(500).json(error.message);
   }
 };
+
+export const mostTavelledDestinations = catchAsync(async (req, res, next) => {
```
Comment (joshwambere) [inline src/controllers/tripRequestController.js:288]: i think you could use one query to get all these data

File: test/tripRequest.test.js [hunk 2]
```diff
@@ -229,4 +229,13 @@ describe('perform CRUD operations on trip request', () => {
         done();
       });
   });
+  it('It should return 200 for most travelled destinations', (done) => {
```
Comment (joshwambere) [inline test/tripRequest.test.js:402]: write negative tests too

**Review by joshwambere — CHANGES_REQUESTED** [review]
refer to the most voted answer here https://stackoverflow.com/questions/37817808/counting-associated-entries-with-sequelize

File: src/controllers/tripRequestController.js (hunk 1 above)
Comment (uwambajeddy) [reply src/controllers/tripRequestController.js:288]: thank you.

File: test/tripRequest.test.js [hunk 3]
```diff
@@ -229,4 +229,22 @@ describe('perform CRUD operations on trip request', () => {
         done();
       });
   });
+  /*   it('It should return 200 for most travelled destinations', (done) => {
```
Comment (Placide123) [inline test/tripRequest.test.js:232]: why this test are commented

**Review by Placide123 — CHANGES_REQUESTED** [review]

File: test/tripRequest.test.js [hunk 4]
```diff
@@ -229,4 +229,22 @@ describe('perform CRUD operations on trip request', () => {
...
+    api
+      .get(`/api/v1/user/trip//most-travelled-destinations`)
+      .set('Authorization', `Bearer ${managerToken}`)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        done();
+      });
+  }); */
```
Comment (Placide123) [inline test/tripRequest.test.js]: why this test is commented on while is testing if your endpoint is working in good condition??

**Review by Placide123 — CHANGES_REQUESTED** [review]

File: test/tripRequest.test.js (hunk 4 above)
Comment (uwambajeddy) [reply test/tripRequest.test.js]: If you tested this PR on your end, you shouldn't have asked this question test first.

**Review by semy-tuyi — DISMISSED** [review]

**Review by joshwambere — APPROVED** [review]

**Pull request merged**